### ISSUE TITLE
Ignore Watched Items Toggle for home-screen widgets doesn't work

### ISSUE THREAD

**Issue by jhs39** [jurialmunkey/skin.arctic.horizon#368]
On the movie home-screen, I use the random movie widget and on the television home-screen, I use the random episodes widget.  The toggle for ignoring watched items doesn't work--whether the toggle is switched on or off all watched items are excluded from the widgets.

**Comment by jurialmunkey** [comment]
What toggle? Doesn't sound like a skin issue.

**Comment by jhs39** [comment]
Settings/Skin/Home/Display/ "Show More..." widget item checked...  When I actually click on the "show more" widget item for a home-screen widget that opens a new window with additional titles that meet the criteria for the widget--for example additional recently added movies or additional random movies.  Clicking up inside the new window displays an options menu that allows the user to choose the view type, sort method, etc.  One of those options is Hide Watched which can be clicked on and off.  If Hide Watched is off then watched items are supposed to be allowed to populate the view window but the toggle doesn't work and watched items are always hidden regardless of user preference.

**Comment by jurialmunkey** [comment]
That's a Kodi setting which hides any visible items in the list that are marked as watched. If the plugin source providing the list doesn't include watched items then those items don't exist to Kodi. Kodi can't magically unhide items that the plugin doesn't provide.

**Comment by jhs39** [comment]
I'm not using any third-party plug-ins.  I'm using the plug-in from Default Widgets/Movies/Random Movies.  The Random Episodes widget is also from the default TV widgets menu.  If the default Kodi widgets aren't seeing items that have been played even though they are supposed to be seeing those items how is that not a bug in your skin?  Where are the default widgets supposed to get this information from so they can function properly?  There is another Random Movies widget located in Video Library/Playlist/Random Movies (Skin Playlist).  That widget also excludes all watched items even when the user setting is to include them.  Are you saying that these widgets won't function properly in any skin because of a bug in Kodi 19?  I am really trying very hard to understand why you don't think this qualifies as a bug.

**Comment by jurialmunkey** [comment]
The Random Movies playlist used for the Defaults widgets specifically has a rule `movies last played not in the last 14 days`. It isn't a bug and is working as designed. The whole intent of that playlist is to discover something new to watch.
https://github.com/jurialmunkey/skin.arctic.horizon/blob/matrix/extras/playlists/RandomMovies.xsp

The Kodi option only shows/hides items already in the list based on their watched status - it never adds any new items. Think of it like this: if I give you a deck of playing cards without any aces and then ask you to show me all the diamond cards, an ace of diamonds won't be in those cards you show me even though an ace of diamonds would be in the full pack.

If you want different rules, smart playlists are very simple to make - just go to `Videos > Playlists > New Smart Playlist` and make one according to the rules that you would like and then add it as a widget.
https://kodi.wiki/view/Smart_playlists

BTW - The Random Movies Skin Playlist is the playlist used by the Random Movies Default Widget so those are identical. It's just that skin shortcuts does an auto discovery of playlists which is why it also shows up in the playlists section.

**Comment by jhs39** [comment]
I have used watched item/watched episode widgets on more than a dozen different Kodi Skins over the past five years across multiple builds of Kodi and I have never encountered the type of behavior occurring on your skin that you claim is completely normal and not a bug.  So let me try and clarify what you are saying since I still don't understand.  Based on the "movies not played in the last 14 days" rule are you saying that 15 days after being played a movie or episode will automatically be eligible to show up in the random widgets?  Or based on your statement that the random items widgets use playlists that never add any new items based on their watched status does that mean watched items will never show up in any of these widgets in spite of an explicit setting that's supposed to allow them to show up and despite the fact that already played items showed up in the random item widgets when the setting was enabled in quite literally every Kodi skin that I ever used before yours?

**Comment by jhs39** [comment]
Either the shortcut to instructions you provided to create smart playlists don't apply to Kodi 19 (the page says they are for Leia) or I'm not smart enough to figure out how to do it.  There are no existing smart playlists in either videos or movies to modify when I follow the instructions step by step and there's no way to look for a folder location where any smart playlists might be if they do actually exist.  I'm not a coder so what you consider easy isn't for somebody like me.  Since you don't consider the way that these widgets function in your skin to be a bug and there's apparently no way for me to change the rules so they actually function the way I want I'll just have to be patient and wait for more Kodi 19 skins to become available.  I'm updating the recommendation I made of your skin on Reddit since the inclusion of the random items widgets was the primary draw for me--it's what I used every day in Aura on Kodi 18.9.  Your skin seemed to offer the functionality that I had been looking for but I was obviously wrong.

**Comment by jurialmunkey** [comment]
You dont even need to make a playlist for random movies. Just point the widget at Movies Titles and set widget sort method to random.

**Comment by jhs39** [comment]
That seems to have worked...thanks!

**Comment by jurialmunkey** [comment]
> I have used watched item/watched episode widgets on more than a dozen different Kodi Skins over the past five years across multiple builds of Kodi and I have never encountered the type of behavior occurring on your skin that you claim is completely normal and not a bug.

Every skin of mine in the last 8 years has used this same playlist for the Random Movies widget, so if you've ever used Arctic Zephyr, Eminence, Fuse Neue, Horizon, or Aura you would have encountered this behaviour.

>  Based on the "movies not played in the last 14 days" rule are you saying that 15 days after being played a movie or episode will automatically be eligible to show up in the random widgets?

Yes, that is what the rules of the playlist are. Each time Kodi loads that playlist it will show items not played in the last 14 days sorted by random.

>  I'm updating the recommendation I made of your skin on Reddit since the inclusion of the random items widgets was the primary draw for me--it's what I used every day in Aura on Kodi 18.9. Your skin seemed to offer the functionality that I had been looking for but I was obviously wrong.

Mate, I wrote Aura and it uses exactly the same playlist for Random Movies:
https://github.com/jurialmunkey/skin.aura/blob/master/extras/playlists/RandomMovies.xsp

**Comment by jurialmunkey** [comment]
Also smart playlists are super simple. You don't need to program anything.

Go to `Videos > Playlists` and choose `New Smart Playlist...`
![Honeyview_screenshot00019](https://user-images.githubusercontent.com/4618719/110258598-8336fa00-7ff7-11eb-99c4-2c4cf6beab7d.jpg)

Then you configure the playlist with whatever rules you want. Here I'm making a playlist for Random 80s Movies.
![Honeyview_screenshot00020](https://user-images.githubusercontent.com/4618719/110258614-9ea20500-7ff7-11eb-9db5-b897cca7ed94.jpg)

Then you click OK and give it a filename to save as. Then go to Home customisation dialog and choose `Video Library > Playlists > Name of your playlist`
![Honeyview_screenshot00021](https://user-images.githubusercontent.com/4618719/110258629-aeb9e480-7ff7-11eb-849d-51e7a9690c1a.jpg)

**Comment by jhs39** [comment]
Thanks for taking the time to post these step-by-step instructions which made the process simple.   I was able to create a custom playlist for played and unplayed random TV episodes (which couldn't be created the same way as the random movie title widget) and it seems to be working the way I intended.  I found a lot of posts on the internet from people trying to create custom playlists for Kodi 19 using the xsp method but couldn't get them to work so I thought it was hopeless.  Thanks again.
